airframe-control/src/main/scala/wvlet/airframe/control/Parallel.scala

```diff
@@ -13,16 +13,35 @@
  */
 package wvlet.airframe.control
 
-import java.util.concurrent.atomic.{AtomicBoolean, AtomicReference}
+import java.util.UUID
+import java.util.concurrent.atomic.{AtomicLong, AtomicReference}
 import java.util.concurrent._
 
-import scala.concurrent.duration.Duration
+import wvlet.airframe.jmx.{JMX, JMXAgent}
+import wvlet.log.LogSupport
+
 import scala.reflect.ClassTag
 
 /**
   * Utilities for parallel execution.
   */
-object Parallel {
+object Parallel extends LogSupport {
+
+  class ParallelExecutionStats(
```

Comment:
We can create a static instance of this stats, using AtomicInteger, AtomicReference for updating metrics from multiple threads. Then expose such metrics with `@JMX(description = "total worker threads in Parallel") def totalWorkerThreads: Int = AtomicInteger.get()`